### PR TITLE
add gitContribute to customize contribution URL

### PR DESCRIPTION
**Pull request by superyyrrzz**
#482 
@chenkennt @vwxyzh @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/docfx.website.themes/default/conceptual.html.primary.js [hunk 1]
```diff
@@ -3,6 +3,6 @@ var common = require('./common.js');
 
 exports.transform = function (model) {
   model._disableToc = model._disableToc || !model._tocPath || (model._navPath === model._tocPath);
-  model.docurl = model.docurl || common.getViewSourceHref(model);
+  model.docurl = model.docurl || common.getViewSourceHref(model, model.gitContribute);
```
Comment (qinezh) [inline src/docfx.website.themes/default/conceptual.html.primary.js]: No need to add `model.gitContribute` as a input parameter?

Comment (superyyrrzz) [inline src/docfx.website.themes/default/conceptual.html.primary.js]: It needs the second parameter, see https://github.com/dotnet/docfx/pull/505/files#diff-5bd22e55ddbb69d733cdb3b7284456b2R37. Thus I can link to a different branch when clicking "Improve this Doc" for conceptual doc.

File: src/docfx/SubCommands/BuildCommand.cs [hunk 2]
```diff
@@ -250,41 +250,41 @@ private static void MergeOptionsToConfig(BuildCommandOptions options, BuildJsonC
...
 
-            if (repoInfoFromBaseDirectory != null && repoInfoFromBaseDirectory.RelativePath != null)
+            if (repoInfoFromBaseDirectory?.RelativePath != null)
             {
                 repoInfoFromBaseDirectory.RelativePath = Path.Combine(repoInfoFromBaseDirectory.RelativePath, DocAsCode.Constants.DefaultOverwriteFolderName);
             }
-            object newFileRepository;
-            if (config.GlobalMetadata.TryGetValue("newFileRepository", out newFileRepository))
+            object gitRespositoryOpenToPublicContributors;
+            if (config.GlobalMetadata.TryGetValue("gitContribute", out gitRespositoryOpenToPublicContributors))
```
Comment (ansyral) [inline src/docfx/SubCommands/BuildCommand.cs]: >gitContribute [](start = 51, length = 13)

do we have documentation to update?

**Comment by ansyral** [comment]
:shipit:

**Comment by vwxyzh** [comment]
✔️

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
:100:

File: src/docfx/SubCommands/BuildCommand.cs (hunk 2 above)
Comment (superyyrrzz) [inline src/docfx/SubCommands/BuildCommand.cs]: Seems document about this topic is missing... will add later.

**Pull request merged**